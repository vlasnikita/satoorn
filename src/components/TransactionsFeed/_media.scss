@media only screen and (max-width: 500px) {
    .Transaction{
        width: 320px;
        height: 90px;

        &__body{
            padding: 5px;
            margin-left: 5px;
            width: calc(100% - 90px);
            height: calc(100% - 10px);
        }

        &__status {
            width: 50px;
            height: 50px;
            border-radius: 4px;
            margin: 10px;
        }

        &__title{
            font-size: 1.4rem;
        }

        &__amount{
            width: 80px;
            height: 40px;
            font-size: 1.6rem;
        }

        &__crumbs{
            height: 36px;
            margin-left: 10px;
        }
    }
}