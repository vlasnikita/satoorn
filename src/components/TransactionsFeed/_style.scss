.TransactionsFeed{
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    overflow: scroll;
    width: 100%;
    margin: 0 auto;
    position: relative;
    height: 100%;
    padding-top: 30px;
}

.Transactions__button_back{
    border-radius: 4px;
    padding: 8px 15px;
    font-size: 1.4rem;
    font-weight: 700;
    background: #fee234;
    height: 20px;
    cursor: pointer;
    box-shadow: 0 3px 0 1px #222;
    text-decoration: none;
    color: #666;
    min-width: 70px;
    text-align: center;
    border: 2px solid #333;
    background: none;
    position: absolute;
    top: 40px;
    left: 60px;
    width: auto;
}

.Transaction{
    width: 460px;
    max-width: calc(100% - 80px);
    height: 120px;
    background: #fff;
    color: #222;
    margin-bottom: 20px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    box-shadow: 0px 1px 10px 0px hsla(0,0%,7%,.1);

    &__body{
        margin-left: 20px;
        padding: 10px 0;
        width: calc(100% - 120px);
        height: calc(100% - 20px);
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: flex-start;
    }

    &__title{
        font-weight: bold;
        font-size: 1.6rem;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
        width: 100%;
    }

    &__info{
        margin: auto 0;
        display: flex;
        flex-direction: row;
        justify-content: flex-start;
        align-items: center;
    }

    &__amount{
        width: 100px;
        height: 50px;
        border: 1px solid #ddd;
        border-radius: 8px;
        font-size: 2.2rem;
        color: #222;
        font-weight: bold;
        display: flex;
        justify-content: center;
        align-items: center;
    }

    &__crumbs{
        height: 50px;
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        align-items: flex-start;
        margin-left: 20px;

    }

    &__crumb{
        color: #222;
        font-weight: bold;
        font-size: 1.2rem;
        display: flex;
        flex-direction: row;
        justify-content: flex-start;
        align-items: center;

        &_large{
            width: 100px;
        }

        &:last-child{
            border-right: none;
        }

        &Label{
            color: #aaa;
            font-weight: normal;
        }
    }

    &__status{
        width: 60px;
        height: 60px;
        border-radius: 4px;
        margin: 30px;
    }
}

@import "/media";