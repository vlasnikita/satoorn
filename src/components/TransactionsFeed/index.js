import React, { Component } from 'react';
import { connect } from 'react-redux';
// import moment from "moment";

import { getTransactions } from "Actions/transactions";
import transaction_statuses from "Constants/transaction_statuses";

class TransactionsFeed extends Component {

    componentDidMount() {
        if(this.props.documentTitle) document.title = this.props.documentTitle

        this.props.handleGetTransactions(this.props.profile.id)
    }

    render() {
        return (
            <div className="TransactionsFeed">
                {Array.isArray(this.props.transactions) && this.props.transactions.map((transaction, i) => (
                    <div className="Transaction" key={transaction.id}>
                        <div className="Transaction__body">
                            <h4 className="Transaction__title">{transaction.challengeName}</h4>
                            <div className="Transaction__info">
                                <div className="Transaction__amount">{transaction.amount}&nbsp;₽</div>
                                <div className="Transaction__crumbs">
                                    <div className="Transaction__crumb">
                                        <span className='Transaction__crumbLabel'>тип:</span>
                                            &nbsp;{transaction_statuses[transaction.transactionStatus].text}
                                    </div>
                                    <div className="Transaction__crumb">
                                        <span className='Transaction__crumbLabel'>обновлено:</span>&nbsp;{transaction.dateUpdated}
                                    </div>
                                </div>
                            </div>
                        </div>
                        <img className="Transaction__status" src={transaction_statuses[transaction.transactionStatus].style} />
                    </div>
                ))}
            </div>
        );
    }
}

const mapStateToProps = state => ({
    transactions: state.transactions.transactions,
    profile: state.profile.profile
});

const mapDispatchToProps = (dispatch) => {
    return {
        handleGetTransactions: id => dispatch(getTransactions(id))
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(TransactionsFeed);