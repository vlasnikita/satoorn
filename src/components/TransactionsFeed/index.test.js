import React from 'react'
import TransactionsFeed from './index'
import renderer from 'react-test-renderer'
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
const initialState = {
    transactions: {
        transactions: require('Mocks/transactions'),
        loading: false,
        loaded: false,
    },
    profile: {
        profile: require('Mocks/streamers')[0],
        loaded: true
    }
}
const store = mockStore(initialState);

it('renders correctly', () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <TransactionsFeed />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});