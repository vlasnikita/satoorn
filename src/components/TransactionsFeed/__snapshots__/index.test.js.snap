// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="TransactionsFeed"
>
  <div
    className="Transaction"
  >
    <div
      className="Transaction__body"
    >
      <h4
        className="Transaction__title"
      >
        Some challenge
      </h4>
      <div
        className="Transaction__info"
      >
        <div
          className="Transaction__amount"
        >
          200
           ₽
        </div>
        <div
          className="Transaction__crumbs"
        >
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              тип:
            </span>
             
            донат
          </div>
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              обновлено:
            </span>
             
          </div>
        </div>
      </div>
    </div>
    <img
      className="Transaction__status"
      src="completed.svg"
    />
  </div>
  <div
    className="Transaction"
  >
    <div
      className="Transaction__body"
    >
      <h4
        className="Transaction__title"
      >
        Some challenge 2
      </h4>
      <div
        className="Transaction__info"
      >
        <div
          className="Transaction__amount"
        >
          -200
           ₽
        </div>
        <div
          className="Transaction__crumbs"
        >
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              тип:
            </span>
             
            донат
          </div>
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              обновлено:
            </span>
             
            2020-02-29T22:18:14.964294
          </div>
        </div>
      </div>
    </div>
    <img
      className="Transaction__status"
      src="pending.svg"
    />
  </div>
  <div
    className="Transaction"
  >
    <div
      className="Transaction__body"
    >
      <h4
        className="Transaction__title"
      >
        Some challenge 3
      </h4>
      <div
        className="Transaction__info"
      >
        <div
          className="Transaction__amount"
        >
          1000
           ₽
        </div>
        <div
          className="Transaction__crumbs"
        >
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              тип:
            </span>
             
            возврат
          </div>
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              обновлено:
            </span>
             
            2020-02-29T22:18:14.964319
          </div>
        </div>
      </div>
    </div>
    <img
      className="Transaction__status"
      src="completed.svg"
    />
  </div>
  <div
    className="Transaction"
  >
    <div
      className="Transaction__body"
    >
      <h4
        className="Transaction__title"
      >
        Some challenge 4
      </h4>
      <div
        className="Transaction__info"
      >
        <div
          className="Transaction__amount"
        >
          1000
           ₽
        </div>
        <div
          className="Transaction__crumbs"
        >
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              тип:
            </span>
             
            возврат
          </div>
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              обновлено:
            </span>
             
            2020-02-29T22:18:14.964326
          </div>
        </div>
      </div>
    </div>
    <img
      className="Transaction__status"
      src="pending.svg"
    />
  </div>
  <div
    className="Transaction"
  >
    <div
      className="Transaction__body"
    >
      <h4
        className="Transaction__title"
      >
        Some challenge 5
      </h4>
      <div
        className="Transaction__info"
      >
        <div
          className="Transaction__amount"
        >
          10000
           ₽
        </div>
        <div
          className="Transaction__crumbs"
        >
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              тип:
            </span>
             
            возврат
          </div>
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              обновлено:
            </span>
             
            2020-02-29T22:18:14.964334
          </div>
        </div>
      </div>
    </div>
    <img
      className="Transaction__status"
      src="rejected.svg"
    />
  </div>
  <div
    className="Transaction"
  >
    <div
      className="Transaction__body"
    >
      <h4
        className="Transaction__title"
      >
        Some challenge 2
      </h4>
      <div
        className="Transaction__info"
      >
        <div
          className="Transaction__amount"
        >
          10500
           ₽
        </div>
        <div
          className="Transaction__crumbs"
        >
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              тип:
            </span>
             
            донат
          </div>
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              обновлено:
            </span>
             
            2020-02-29T22:18:14.96431
          </div>
        </div>
      </div>
    </div>
    <img
      className="Transaction__status"
      src="rejected.svg"
    />
  </div>
  <div
    className="Transaction"
  >
    <div
      className="Transaction__body"
    >
      <h4
        className="Transaction__title"
      >
        Some challenge 2
      </h4>
      <div
        className="Transaction__info"
      >
        <div
          className="Transaction__amount"
        >
          10500
           ₽
        </div>
        <div
          className="Transaction__crumbs"
        >
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              тип:
            </span>
             
            выплата
          </div>
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              обновлено:
            </span>
             
            2020-02-29T22:18:14.96431
          </div>
        </div>
      </div>
    </div>
    <img
      className="Transaction__status"
      src="pending.svg"
    />
  </div>
  <div
    className="Transaction"
  >
    <div
      className="Transaction__body"
    >
      <h4
        className="Transaction__title"
      >
        Some challenge 2
      </h4>
      <div
        className="Transaction__info"
      >
        <div
          className="Transaction__amount"
        >
          10500
           ₽
        </div>
        <div
          className="Transaction__crumbs"
        >
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              тип:
            </span>
             
            выплата
          </div>
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              обновлено:
            </span>
             
            2020-02-29T22:18:14.96431
          </div>
        </div>
      </div>
    </div>
    <img
      className="Transaction__status"
      src="rejected.svg"
    />
  </div>
  <div
    className="Transaction"
  >
    <div
      className="Transaction__body"
    >
      <h4
        className="Transaction__title"
      >
        Some challenge 2
      </h4>
      <div
        className="Transaction__info"
      >
        <div
          className="Transaction__amount"
        >
          10500
           ₽
        </div>
        <div
          className="Transaction__crumbs"
        >
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              тип:
            </span>
             
            выплата
          </div>
          <div
            className="Transaction__crumb"
          >
            <span
              className="Transaction__crumbLabel"
            >
              обновлено:
            </span>
             
            2020-02-29T22:18:14.96431
          </div>
        </div>
      </div>
    </div>
    <img
      className="Transaction__status"
      src="completed.svg"
    />
  </div>
</div>
`;
